### PR TITLE
Keystore persistence

### PR DESCRIPTION
**Pull request by neekolas**
## Summary

- Start implementing the basic persistence layer for the Keystore. Wanted to get the building block pieces in before I start work on the encryption layer.
- Includes a LocalStorage polyfill that doesn't rely on the filesystem or pollute globals, to avoid past pains with off-the-shelf polyfills.

## Next steps
- Implement encrypted persistence
- Implement conversation storage using encrypted persistence class

## Notes
- A lot of this code is very similar to what lives in `src/store`. The store folder is ultimately going to be deleted once the keystore bootstrapping part of V8 comes in. Some of the code will make its way into the keystore, which will resolve some of the naming confusion (`store` has a class named `KeyStore` which will be integrated into the new Keystore and renamed to something less confusing)

### CODE REVIEW

File: src/keystore/persistence/LocalStoragePonyfill.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// Fully in-memory polyfill for the browser storage API.
+// Borrowed from https://github.com/MitchellCash/node-storage-polyfill but implemented as a ponyfill instead of a polyfill
+
+export default class LocalStoragePonyfill implements Storage {
```
Comment (michaelx11) [inline src/keystore/persistence/LocalStoragePonyfill.ts:4]: TIL: ponyfill is a thing

File: src/keystore/persistence/LocalStoragePonyfill.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  store: Map<string, string>
+  constructor() {
+    this.store = new Map()
+  }
+
+  get length(): number {
+    return this.store.size
```
Comment (michaelx11) [inline src/keystore/persistence/LocalStoragePonyfill.ts:11]: I have the worst suggestion ever: Would it be worth tracking a `total_bytes` parameter and having a `byte_size` method? Might be useful for metrics and stuff but probably not worth the bookkeeping right now.

**Review by michaelx11 — APPROVED** [review]
lgtm

File: src/keystore/persistence/LocalStoragePonyfill.ts (hunk 2 above)
Comment (neekolas) [reply src/keystore/persistence/LocalStoragePonyfill.ts:11]: So, this ponyfill is meant to be a 1:1 match of the browser `Storage` API.

If we wanted to track `total_bytes` I'd be tempted to do that bookkeeping on top of the real `LocalStorage`, which does have some hard limits (5mb total).

**Review by jazzz — APPROVED** [review]
🚀

File: src/keystore/persistence/LocalStoragePonyfill.ts (hunk 2 above)
Comment (michaelx11) [reply src/keystore/persistence/LocalStoragePonyfill.ts:11]: Oh that makes a lot more sense

**Pull request merged**